Content.Server/Atmos/GasSprayerComponent.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.Interfaces;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Pointing;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+
+
+namespace Content.Server.Atmos
+{
+    [RegisterComponent]
+    public class GasSprayerComponent : Component, IAfterInteract
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager = default!;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager = default!;
+#pragma warning restore 649
+
+        //TODO: create a function that can create a gas based on a solution mix
+        public override string Name => "GasSprayer";
+
+        private string _spraySound;
+        private string _sprayType;
+        private string _fuelType;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _spraySound, "spraySound", string.Empty);
+            serializer.DataField(ref _sprayType, "sprayType", string.Empty);
+            serializer.DataField(ref _fuelType, "fuelType", string.Empty);
+
+        }
+
+        public void AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (Owner.TryGetComponent(out SolutionComponent tank) &&
+                tank.Solution.GetReagentQuantity(_fuelType).Float().Equals(0f))
+            {
+                _notifyManager.PopupMessage(Owner, eventArgs.User,
+                    Loc.GetString("The {0} is out of {1}!", Owner.Name, _fuelType));
+            }
+            else
+            {
+                tank.TryRemoveReagent(_fuelType, ReagentUnit.New(50));
+
+                var playerPos = eventArgs.User.Transform.GridPosition;
+                var direction = (eventArgs.ClickLocation.Position - playerPos.Position).Normalized;
+                playerPos.Offset(direction);
```

Comment:
This suffers from the same "problem" as my VaporComponent. I don't know if just offsetting it one tile good enough.

Reply:
Could you elaborate? What would be a "correct" offset?

Reply:
I think the best way would be to spawn it directly in front of the player, so that it doesn't touch him.
This would basically be playerPos + The AABB "radius" (which would be 0.5?).

Reply:
Went ahead and just cut the offset in half. Worked without any issues.